specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2022-08-03/communityGallery.json

```diff
@@ -0,0 +1,609 @@
+        "description": "List community gallery image versions inside an image.",
+        "parameters": [
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./galleryRPCommon.json#/parameters/LocationNameParameter"
+          },
+          {
+            "$ref": "../../../common-types/v1/common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/PublicGalleryNameParameter"
+          },
+          {
+            "$ref": "#/parameters/GalleryImageNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/CommunityGalleryImageVersionList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common-types/v1/common.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List community gallery image versions.": {
+            "$ref": "./examples/communityGalleryExamples/CommunityGalleryImageVersion_List.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "PirCommunityGalleryResource": {
+      "properties": {
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource name"
+        },
+        "location": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource location"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type"
+        },
+        "identifier": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/CommunityGalleryIdentifier"
+        }
+      },
+      "description": "Base information about the community gallery resource in pir."
+    },
+    "CommunityGalleryArtifactTags": {
+      "type": "object",
+      "additionalProperties": {
+        "type": "string"
+      },
+      "description": "The artifact tags of a community gallery resource."
+    },
+    "CommunityGalleryDisclaimer": {
+      "type": "string",
+      "properties": {
+        "disclaimer": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Community gallery disclaimer"
+        }
+      },
+      "description": "The disclaimer for a community gallery resource."
+    },
+    "CommunityGalleryIdentifier": {
+      "properties": {
+        "uniqueId": {
+          "type": "string",
+          "description": "The unique id of this community gallery."
+        }
+      },
+      "description": "The identifier information of community gallery."
+    },
+    "CommunityGalleryMetadata": {
```

Comment:
>CommunityGalleryMetadata [](http://example.com/codeflow?start=5&length=24)

are any required?

Reply:
None of them is required. CommunityGalleryMetadata itself is null-able.

Reply:
I'm just trying to visualize the scenario here, and let me know if this seems misguided but..
I just feel like, if I was going to use a package, I would want to know a couple basic facts at a minimum. Like who claims to be the author, and how they claim to be licensing me the package. So, I wondered why these are optional.

Reply:
oh you're right, this is required in communityGallery.json, not in gallery.json. Let me update it.

Reply:
added

Reply:
@kangsun-ctrl Sorry I seem to be missing something. What did you add?